dev/src/order.ts

```diff
@@ -248,19 +248,38 @@ function compareVectors(left: ApiMapValue, right: ApiMapValue): number {
   return compareArrays(leftArray, rightArray);
 }
 
-function stringToUtf8Bytes(str: string): Uint8Array {
-  return new TextEncoder().encode(str);
-}
-
 /*!
  * Compare strings in UTF-8 encoded byte order
  * @private
  * @internal
  */
 export function compareUtf8Strings(left: string, right: string): number {
-  const leftBytes = stringToUtf8Bytes(left);
-  const rightBytes = stringToUtf8Bytes(right);
-  return compareBlobs(Buffer.from(leftBytes), Buffer.from(rightBytes));
+  let i = 0;
+  while (i < left.length && i < right.length) {
+    const leftCodePoint = left.codePointAt(i)!;
+    const rightCodePoint = right.codePointAt(i)!;
+
+    if (leftCodePoint !== rightCodePoint) {
+      if (leftCodePoint < 128 && rightCodePoint < 128) {
```

Comment:
Consider making the body of this outer "if" statement a single line, and factor out the logic to do the actual comparison to a helper function. IMO this will improve readability because, from a quick code inspection, it's not obvious that the function is guaranteed to return if the "if" block condition is satisfied.

I'm thinking something like this:

```ts
if (leftCodePoint !== rightCodePoint) {
  return isLeftGreaterAt(left, right, i) ? 1 : -1;
}
```

This also makes it clear that zero is _never_ a potential return value once the code enters this "if" block.

Reply:
(double-check my logic there... I did not test the code!)

Reply:
this down the block somehow conveys the same message
```      
if (comp !== 0) {
    return comp;
}
```

`isLeftGreaterAt` returning true/false duplicates the logic where we check if the compare result is 1 or -1 inside the function, and we check the `isLeftGreaterAt` result again to decide return 1 or -1.

Reply:
I understand that the ascii set is 0-127, but the cutoff for one-char vs two-char code points is `0xffff`. Should this condition be changed to

```
if (leftCodePoint < 0xffff && rightCodePoint < 0xffff) {
```
?

Reply:
Consider omitting this special treatment of code points less than 128 in this PR. This special treatment is effectively an _optimization_, and I would suggest implementing optimizations like this in a follow-up PR, once correctness has been established.